sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -461,68 +462,269 @@ case class FindInSet(left: Expression, right: Expression) extends BinaryExpressi
 }
 
 /**
- * A function that trim the spaces from both ends for the specified string.
- */
+ * A function that takes a character string, removes the leading and/or trailing characters matching with the characters
+ * in the trim string, returns the new string. If LEADING/TRAILING/BOTH and trimStr keywords are not specified, it
+ * defaults to remove space character from both ends.
+ * trimStr: A character string to be trimmed from the source string, if it has multiple characters, the function
+ * searches for each character in the source string, removes the characters from the source string until it
+ * encounters the first non-match character.
+ * LEADING: removes any characters from the left end of the source string that matches characters in the trim string.
+ * TRAILING: removes any characters from the right end of the source string that matches characters in the trim string.
+ * BOTH: removes any characters from both ends of the source string that matches characters in the trim string.
+  */
 @ExpressionDescription(
-  usage = "_FUNC_(str) - Removes the leading and trailing space characters from `str`.",
+  usage = """
+    _FUNC_(str) - Removes the leading and trailing space characters from `str`.
+    _FUNC_(BOTH trimStr FROM str) - Remove the leading and trailing trimString from `str`
```

Comment:
yes, it is only for "BOTH", I was thinking that the user can specify TRAILING and LEADING through trim, so I put here. But it may cause confusion. I will remove these two. Thanks.